AppSearchSample/app/src/main/java/com/android/example/appsearchsample/NoteListItemAdapter.kt

```diff
@@ -43,31 +52,44 @@ class NoteListItemAdapter(private val onDelete: (Note?) -> Unit) :
   }
 
   override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
-    holder.bind(getItem(position), onDelete)
+    holder.bind(getItem(position), onDelete, position)
   }
 
   /** ViewHolder for [NoteListItemAdapter]. */
-  class NoteViewHolder(
+  inner class NoteViewHolder(
     binding: ItemNoteBinding,
   ) : RecyclerView.ViewHolder(binding.root) {
     private val noteTextView = binding.noteText
     private val noteDeleteButtonView = binding.noteDeleteButton
 
-    fun bind(note: Note?, onDelete: (Note?) -> Unit) {
-      noteTextView.text = note?.text
-      noteDeleteButtonView.setOnClickListener { onDelete(note) }
+    fun bind(sr: SearchResult, onDelete: (SearchResult?) -> Unit, position: Int) {
+      val note = sr.genericDocument.toDocumentClass(Note::class.java)
+      val sb = SpannableStringBuilder(note.text)
+
+      sr.matchInfos.forEach {
```

Comment:
To demonstrate best practices to devs looking at the sample, we should also check that we are only bolding the intended property path (in this case "text") for each MatchInfo.

It doesn't make a difference in this example, as we have only have one indexed path so all matches should be for the "text" property, but other uses may need this check.